src/GraphQL/Server.hs

```diff
@@ -37,18 +38,16 @@ import GraphQL.Internal.Schema (HasName(..))
 import GraphQL.API
 import GraphQL.Internal.Input (CanonicalQuery)
 
-import Control.Monad.Catch (MonadThrow, throwM, Exception)
+import Control.Monad.Catch (Exception)
+
+data QueryError =
```

Comment:
I'd like us to rename this type now that it represents more than just query errors. I think `Error` would be a fine name, although `GraphQLError` would also be ok.